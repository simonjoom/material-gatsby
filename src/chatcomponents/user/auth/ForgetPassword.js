import React, { Component } from "react";
import { graphql, compose } from "react-apollo";
import gql from "graphql-tag";
import Button from "../../../reactLIB/Button";
var validator = require("email-validator");

class ForgetPassword extends Component {
  state = {
    email: ""
  };
  notifyAboutComment(comment) {
    var toastHTML =
      "<span>" +
      comment +
      "</span><button class='btn-flat toast-action'>Undo</button>";
    typeof M !== "undefined" && M.toast({ html: toastHTML });
  }
  validateEmail(email) {
    this.pass = validator.validate(email);
    return this.pass;
  }

  render() {
    return (
      <div className="paperOut">
        <Card className="paperIn" s={12}>
          <h4 className="mv3">Forget Password</h4>
          <div className="md-grid">
            <Input
              value={this.state.email}
              success={this.validateEmail(this.state.email)}
              type="email"
              onChange={e => this.setState({ email: e.target.value })}
              label="Your email address"
              s={12}
            />
          </div>
          <div className="flex mt3">
            <Button onClick={() => this._confirm()}>Ok</Button>
          </div>
        </Card>
      </div>
    );
  }

  _confirm = async () => {
    const { email } = this.state;
    let messageSnackBar;
    await this.props
      .forgetPasswordMutation({
        variables: {
          email
        }
      })
      .then(result => {
        messageSnackBar = `A mail has been sent with a link available until
        ${new Date(
          result.data.forgetPassword.resetPasswordExpires
        ).toLocaleString()}`;
        this.notifyAboutComment(messageSnackBar);
      })
      .catch(e => {
        this.notifyAboutComment(e.graphQLErrors[0].message);
      });
  };
}

const FORGET_PASSWORD_MUTATION = gql`
  mutation ForgetPasswordMutation($email: String!) {
    forgetPassword(email: $email) {
      name
      id
      resetPasswordExpires
    }
  }
`;

export default compose(
  graphql(FORGET_PASSWORD_MUTATION, { name: "forgetPasswordMutation" })
)(ForgetPassword);
