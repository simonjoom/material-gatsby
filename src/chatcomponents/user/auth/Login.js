import React, { Component } from "react";
import { AUTH_TOKEN } from "../../../constants/constants";
import { graphql, compose, withApollo } from "react-apollo";
import { navigate } from "gatsby";
import gql from "graphql-tag"; 
import Button from "../../../reactLIB/Button";
import Card from "../../../reactLIB/Card";
import Input from "../../../reactLIB/Input"; 

export class Login extends Component {
  constructor(props) {
    super(props)

    this.state = {
      email: null,
      password: "",
      name: "",
      loading: false,
    }
 
    this.validateEmail = this.validateEmail.bind(this)
    this.validatePassword = this.validatePassword.bind(this)
  }

  notifyAboutComment(comment) {
    var toastHTML =
      "<span>" +
      comment +
      "</span><button class='btn-flat toast-action'>Undo</button>";
    typeof M !== "undefined" && M.toast({ html: toastHTML });
  }
  /*validateEmail(email) {
    var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
   console.log("valide",re.test(email))
    return re.test(email);
  }*/
  validateEmail() {
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(this.state.email)
  }

  validatePassword() {
    return !!this.state.password
  }

  render() {
    return (
      <div className="paperOut">
        <Card>
          <h4 className="mv3">Login</h4>
          <div className="md-grid" style={{ width: "100%", marginTop: 20 }}>
            <Input
              id="email"
              value={this.state.email}
              onChange={e => this.setState({ email: e.target.value })}
              type="email"
              validate
              success={this.validateEmail(this.state.email)}
              label="Email"
              placeholder="Your email address"
              required
              s={12}
            />
            <Input
              id="password"
              value={this.state.password}
              onChange={e => this.setState({ password: e.target.value })}
              type="password"
              validate
              success={this.validatePassword()}
              label="Password"
              placeholder="Your Password"
              required
              s={12}
            />
          </div>
          <div
            className="flex"
            style={{ display: "flex", justifyContent: "center" }}
          >
            <Button
              className="btn btn-small"
              id="ok"
              onClick={() => this._confirm()}
              style={{ margin: "5%" }}
            >
              Signin
            </Button>

            <Button
              className="btn btn-small"
              onClick={() => navigate("/z/signup")}
              style={{ margin: "5%" }}
            >
              signup
            </Button>
          </div>
          <div
            className="flex"
            style={{ display: "flex", justifyContent: "center" }}
          >
            <Button
              className="btn btn-small"
              onClick={() => navigate("/z/forgetPassword")}
              style={{ margin: "2%" }}
            >
              Forgot Password
            </Button>
          </div>
        </Card>
      </div>
    );
  }

  _confirm = async () => {
    const { email, password } = this.state;

    await this.props
      .loginMutation({
        variables: {
          email,
          password
        }
      })
      .then(result => {
        const { token, user } = result.data.login;
        this._saveUserData(token, user);
        // this.props.history.push(`/`)
        // window.location.reload()
      })
      .catch(e => { 
        this.notifyAboutComment(e.graphQLErrors[0].message);
      });
  };

  _saveUserData = (token, user) => {
    localStorage.setItem(AUTH_TOKEN, token);
    localStorage.setItem("userToken", JSON.stringify(user));
    this.props.client.resetStore().then(() => {
      navigate(`/`);
    });
  };
}

const LOGIN_MUTATION = gql`
  mutation LoginMutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        name
        id
      }
    }
  }
`;

export default compose(
  withApollo,
  graphql(LOGIN_MUTATION, { name: "loginMutation" })
)(Login);
