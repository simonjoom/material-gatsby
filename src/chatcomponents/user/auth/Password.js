import React, { Component } from "react";
import Input from "../../../reactLIB/Input";
import Button from "../../../reactLIB/Button";

export default class Password extends Component {
  state = {
    inputValidation2: false,
    isPasswordLongEnough: true,
    hasLowerCase: true,
    hasUpperCase: true,
    showPassword: false,
    // hasNumber: true,
    // hasSpecialChar: true,
    password: "",
    passwordMinimumLength: 8
  };

  UNSAFE_componentWillReceiveProps() {
    if (!this.state.inputValidation2) this.inputRef.focus();
  }
  onChange(e) {
    let inputValidation2 = false;
    if (
      this.isPasswordLongEnough(e.target.value) &&
      this.hasLowerCase(e.target.value) &&
      this.hasUpperCase(e.target.value)
      //  this.hasNumber(e.target.value) &&
      //  this.hasSpecialChar(e.target.value)
    ) {
      inputValidation2 = true;
    }

    this.setState(
      {
        password: e.target.value,
        inputValidation2: inputValidation2,
        //  hasNumber: this.hasNumber(e.target.value),
        //  hasSpecialChar: this.hasSpecialChar(e.target.value),
        hasUpperCase: this.hasUpperCase(e.target.value),
        hasLowerCase: this.hasLowerCase(e.target.value),
        isPasswordLongEnough: this.isPasswordLongEnough(e.target.value)
      },
      () => {
        this.props.onChange(this.state);
      }
    );
  }

  hasLowerCase(str) {
    return str.toUpperCase() !== str;
  }
  hasUpperCase(str) {
    return str.toLowerCase() !== str;
  }
  hasNumber(string) {
    return /\d/.test(string);
  }

  hasSpecialChar(str) {
    var format = /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]+/;
    if (format.test(str)) {
      return true;
    } else {
      return false;
    }
  }

  isPasswordLongEnough(password) {
    if (password.length > this.state.passwordMinimumLength) {
      this.setState({ isPasswordLongEnough: true });
      return true;
    }
    this.setState({ isPasswordLongEnough: false });
    return false;
  }

  handleNext = () => {
    this.props.handleNext();
  };

  handleKey = data => {
    if (data.charCode === 13) {
      //keyPress enter
      this.handleNext();
    }
  };
  showPassword() {
    this.setState({ showPassword: !this.state.showPassword });
  }
  render() {
    const ButtonStart = () => (
      <Button
        onClick={() => this.showPassword()}
        type="material"
        icon={this.state.showPassword ? "visibility_off" : "visibility"}
        flat
      />
    );
    const { placeholder, id, label } = this.props;
    const ButtonNext = this.props.ButtonNext
      ? this.props.ButtonNext
      : () => <div />;
    const errormessage = !this.state.isPasswordLongEnough
      ? "At least " + this.state.passwordMinimumLength + " characters long."
      : !this.state.hasLowerCase
        ? "At least a lower case letter."
        : !this.state.hasUpperCase
          ? "At least an upper case letter."
          : "";
    return (
      <div className="wrapperAnimate focusField">
        <Input
          id={id}
          value={this.state.password}
          label={label}
          placeholder={placeholder}
          errormessage={errormessage}
          success={this.state.inputValidation2}
          onChange={this.onChange.bind(this)}
          type={this.state.showPassword ? "text" : "password"}
          setRef={ref => {
            this.inputRef = ref;
            this.props.setRef && this.props.setRef(ref);
          }}
          onKeyPress={this.handleKey}
          inline
          buttonIcon={<ButtonNext icon={this.props.nextIcon} />}
          buttonIconStart={<ButtonStart />}
          s={12}
        />
      </div>
    );
  }
}

Password.defaultProps = {
  id: "password",
  label: "Password",
  placeholder: "Choose a safe password"
};

/*
    
      {!this.state.hasNumber && (
        <FormHelperText>At least an number.</FormHelperText>
      )} {!this.state.hasSpecialChar && (
        <FormHelperText>At least a spceial character.</FormHelperText>
      )}*/
