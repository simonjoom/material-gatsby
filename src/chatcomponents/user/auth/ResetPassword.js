import React, { Component } from "react";
import { AUTH_TOKEN } from "../../../constants/constants";
import { graphql, compose } from "react-apollo";
import gql from "graphql-tag";
import Password from "./Password";
import Button from "../../../reactLIB/Button";
import Card from "../../../reactLIB/Card";
import Input from "../../../reactLIB/Input";

const queryString = require("query-string");

class ResetPassword extends Component {
  state = {
    password: "",
    password2: "",
    inputValidation2: true,
    resetPasswordToken: ""
  };

  notifyAboutComment(comment) {
    var toastHTML =
      "<span>" +
      comment +
      "</span><button class='btn-flat toast-action'>Undo</button>";
    typeof M !== "undefined" && M.toast({ html: toastHTML });
  }

  componentDidMount() {
    let resetPasswordToken = queryString.parse(this.props.location.search)
      .resetPasswordToken;
    if (resetPasswordToken) {
      this.setState({
        resetPasswordToken: resetPasswordToken
      });
    }
  }

  onChange(statePasword) { 
    this.setState({
      password: statePasword.password,
      inputValidation2: statePasword.inputValidation2
    });
  }

  handleNext = () => {
    if (!this.state.inputValidation2) {
      // alert("correct password");
    }
  };

  render() {
    return (
      <div className="paperOut">
        <Card>
          <h4 className="mv3">Reset Password</h4>
          <div className="md-grid">
            <Password
              handleNext={this.handleNext.bind(this)}
              onChange={this.onChange.bind(this)}
              setRef={this.setRef}
            />

            <Input
              s={12}
              value={this.state.password2}
              onChange={e => this.setState({ password2: e.target.value })}
              type="password"
              errormessage="Error: Passwords are differents"
              success={this.state.password === this.state.password2}
              label="Retype your safe password"
            />
          </div>
          <div className="flex mt3">
            <Button onClick={() => this._confirm()}>Ok</Button>
          </div>
        </Card>
      </div>
    );
  }

  _confirm = async () => {
    if (!this.state.inputValidation2) {
      return;
    }
    if (this.state.password !== this.state.password2) {
      this.notifyAboutComment("Error: Passwords are differents");
      return;
    }
    const { password, resetPasswordToken } = this.state;
    await this.props
      .resetPasswordMutation({
        variables: {
          resetPasswordToken,
          password
        }
      })
      .then(result => {
        this.notifyAboutComment("Your password has been reset successfully!");
        const { token, user } = result.data.resetPassword;
        this._saveUserData(token, user);
      })
      .catch(e => {
        this.notifyAboutComment(e.graphQLErrors[0].message);
      });
  };

  _saveUserData = (token, user) => {
    localStorage.setItem(AUTH_TOKEN, token);
    localStorage.setItem("userToken", JSON.stringify(user));
  };
}

const RESET_PASSWORD_MUTATION = gql`
  mutation ResetPasswordMutation(
    $password: String!
    $resetPasswordToken: String!
  ) {
    resetPassword(
      password: $password
      resetPasswordToken: $resetPasswordToken
    ) {
      token
      user {
        name
        id
      }
    }
  }
`;

export default compose(
  graphql(RESET_PASSWORD_MUTATION, { name: "resetPasswordMutation" })
)(ResetPassword);
