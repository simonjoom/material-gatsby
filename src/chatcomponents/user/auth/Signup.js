import React, { Component } from "react";
import { AUTH_TOKEN } from "../../../constants/constants";
import { graphql, compose } from "react-apollo";
import { navigate } from "gatsby";
import gql from "graphql-tag";
import Password from "./Password";
//import Button from '@material-ui/core/Button'
import { withApollo } from "react-apollo";
import ProgressBar from "../../../reactLIB/ProgressBar";
import Card from "../../../reactLIB/Card";
import Button from "../../../reactLIB/Button";
import Input from "../../../reactLIB/Input";

var validator = require("email-validator");

class Signup extends Component {
  state = {
    email: "",
    emailValidation: true,
    inputValidation2: true,
    password: "",
    name: "",
    isPasswordActiveStep: false,
    nameFile: "",
    activeStep: 0,
    maxStep: 3
  };

  notifyAboutComment(comment) {
    var toastHTML =
      "<span>" +
      comment +
      "</span><button class='btn-flat toast-action'>Undo</button>";
    typeof M !== "undefined" && M.toast({ html: toastHTML });
  }

  onChange(statePasword) {
    this.setState({
      password: statePasword.password,
      inputValidation2: statePasword.inputValidation2
    });
  }

  validateEmail(email) {
    this.pass = validator.validate(email);
    return this.pass;
  }
  validateName(name) {
    this.pass = name.length > 0;
    return this.pass;
  }
  calculateBuffer() {
    let data = "";
    if (this.state.activeStep === 0) {
      data = this.state.name;
    }
    if (this.state.activeStep === 1) {
      data = this.state.email;
    }
    if (this.state.activeStep === 2) {
      data = this.state.password;
    }
    let maxValue = data.length / 10 > 1 ? 1 : data.length / 10;
    return ((this.state.activeStep + maxValue) * 100) / this.state.maxStep;
  }
  handleNext = () => {
    if (this.state.name) {
      if (this.state.activeStep === 0) {
        this.setState(
          {
            activeStep: this.state.activeStep + 1
          },
          () => {
            this.inputRef.focus();
          }
        );
      }
      if (this.state.activeStep === 1) {
        if (this.state.emailValidation) {
          this.setState(
            {
              activeStep: this.state.activeStep + 1
            },
            () => {
              this.inputRef.focus();
              this.setState({ isPasswordActiveStep: true });
            }
          );
        }
      }
      if (this.state.activeStep === 2) {
        if (this.state.inputValidation2) {
          this.setState(
            {
              activeStep: this.state.activeStep + 1
            },
            () => {
              this._confirm();
            }
          );
        }
      }
    }
  };

  handleKey = data => {
    if (data.charCode === 13 && this.pass) {
      //keyPress enter
      this.handleNext();
    }
  };
  setRef = ref => {
    this.inputRef = ref;
  };
  render() {
    const ButtonNext = ({ index, icon = "navigate_next" }) =>
      this.state.activeStep === index ? (
        <Button
          onClick={this.handleNext}
          type="material"
          icon={icon}
          floating
        />
      ) : null;
    return (
      <div className="paperOut">
        <Card>
          <h4 className="mv3">Sign Up</h4>
          <div className="flex flex-column">
            <ProgressBar
              variant="buffer"
              progress={(this.state.activeStep * 100) / this.state.maxStep}
              valueBuffer={this.calculateBuffer()}
            />
            <br />
            <div className="tac">
              <div
                className={
                  "wrapperAnimate " +
                  (this.state.activeStep === 0 ? "focusField" : "notFocusField")
                }
              >
                <Input
                  id="name"
                  label="Your name"
                  placeholder="Your name"
                  value={this.state.name}
                  success={this.validateName(this.state.name)}
                  setRef={this.setRef}
                  onChange={e => this.setState({ name: e.target.value })}
                  type="text"
                  onKeyPress={this.handleKey}
                  inline
                  buttonIcon={<ButtonNext index={0} />}
                  s={12}
                />
              </div>

              <br />
              {this.state.activeStep >= 1 && (
                <div
                  className={
                    "wrapperAnimate " +
                    (this.state.activeStep === 1
                      ? "focusField"
                      : "notFocusField")
                  }
                >
                  <Input
                    id="email"
                    label="Email"
                    placeholder="Your email address"
                    value={this.state.email}
                    success={this.validateEmail(this.state.email)}
                    onChange={e => this.setState({ email: e.target.value })}
                    setRef={this.setRef}
                    type="email"
                    onKeyPress={this.handleKey}
                    inline
                    buttonIcon={<ButtonNext index={1} />}
                    s={12}
                  />
                </div>
              )}

              <br />
              <br />
              {this.state.activeStep >= 2 && (
                <Password
                  handleNext={this.handleNext.bind(this)}
                  onChange={this.onChange.bind(this)}
                  ButtonNext={ButtonNext}
                />
              )}
            </div>
          </div>
        </Card>
      </div>
    );
  }

  _confirm = async () => {
    const { name, email, password, nameFile } = this.state;
    await this.props
      .signupMutation({
        variables: {
          name,
          email,
          password,
          nameFile
        }
      })
      .then(result => {
        const { token, user } = result.data.signup;
        this._saveUserData(token, user);
      })
      .catch(e => {
        if (e.graphQLErrors.length) {
          this.notifyAboutComment(e.graphQLErrors[0].message);
        }
      });
  };

  _saveUserData = (token, user) => {
    localStorage.setItem(AUTH_TOKEN, token);
    localStorage.setItem("userToken", JSON.stringify(user));
    this.props.client.resetStore().then(() => {
      navigate(`/`);
    });
  };
}

const SIGNUP_MUTATION = gql`
  mutation SignupMutation(
    $email: String!
    $password: String!
    $name: String!
    $nameFile: String!
  ) {
    signup(
      email: $email
      password: $password
      name: $name
      nameFile: $nameFile
    ) {
      token
      user {
        name
        id
      }
    }
  }
`;

export default compose(
  withApollo,
  graphql(SIGNUP_MUTATION, { name: "signupMutation" })
)(Signup);
