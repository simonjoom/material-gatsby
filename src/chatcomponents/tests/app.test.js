import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import Signup from '../user/auth/Signup'
import { Login } from '../user/auth/Login'

configure({ adapter: new Adapter() })

it('Signup', () => {
  const wrapper = shallow(
    <Signup/>
  )
  expect(wrapper).toMatchSnapshot()
})

it('Login', () => {
  const wrapper = shallow(
    <Login/>
  )
  expect(wrapper).toMatchSnapshot()
})

it('renders a email input', () => {
  const wrapper = shallow(<Login />)
  expect(wrapper.find('#email').length).toEqual(1)
})

it('email input', () => {
  const wrapper = shallow(<Login />)
  wrapper.find('#email').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}})
  expect(wrapper.state('email')).toEqual('blah@gmail.com')
})

describe('Password input', () => {
  it('should respond to change event and change the state of the Login Component', () => {
    const wrapper = shallow(<Login />)
    wrapper.find('#password').simulate('change', {target: {name: 'password', value: 'cats'}})
    expect(wrapper.state('password')).toEqual('cats')
  })
})
