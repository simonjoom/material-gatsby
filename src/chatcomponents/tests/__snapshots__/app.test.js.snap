// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Paper)
        className="paperIn"
>
        <h4
                className="mv3"
        >
                Login
        </h4>
        <div
                className="flex flex-column"
        >
                <TextField
                        id="email"
                        label="Your email address"
                        onChange={[Function]}
                        required={false}
                        select={false}
                        type="text"
                        value=""
                />
                <TextField
                        id="password"
                        label="Password"
                        onChange={[Function]}
                        required={false}
                        select={false}
                        type="password"
                        value=""
                />
        </div>
        <div
                className="flex mt3"
        >
                <WithStyles(Button)
                        id="ok"
                        onClick={[Function]}
                        variant="raised"
                >
                        Ok
                </WithStyles(Button)>
                <WithStyles(Button)
                        onClick={[Function]}
                        variant="flat"
                >
                        signup
                </WithStyles(Button)>
                <WithStyles(Button)
                        onClick={[Function]}
                        variant="flat"
                >
                        Forget Password
                </WithStyles(Button)>
        </div>
        <SnackBarCustom />
</WithStyles(Paper)>,
      "className": "paperOut",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h4
            className="mv3"
>
            Login
</h4>,
          <div
            className="flex flex-column"
>
            <TextField
                        id="email"
                        label="Your email address"
                        onChange={[Function]}
                        required={false}
                        select={false}
                        type="text"
                        value=""
            />
            <TextField
                        id="password"
                        label="Password"
                        onChange={[Function]}
                        required={false}
                        select={false}
                        type="password"
                        value=""
            />
</div>,
          <div
            className="flex mt3"
>
            <WithStyles(Button)
                        id="ok"
                        onClick={[Function]}
                        variant="raised"
            >
                        Ok
            </WithStyles(Button)>
            <WithStyles(Button)
                        onClick={[Function]}
                        variant="flat"
            >
                        signup
            </WithStyles(Button)>
            <WithStyles(Button)
                        onClick={[Function]}
                        variant="flat"
            >
                        Forget Password
            </WithStyles(Button)>
</div>,
          <SnackBarCustom />,
        ],
        "className": "paperIn",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "mv3",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextField
                id="email"
                label="Your email address"
                onChange={[Function]}
                required={false}
                select={false}
                type="text"
                value=""
/>,
              <TextField
                id="password"
                label="Password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                value=""
/>,
            ],
            "className": "flex flex-column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "email",
                "label": "Your email address",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "password",
                "label": "Password",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                id="ok"
                onClick={[Function]}
                variant="raised"
>
                Ok
</WithStyles(Button)>,
              <WithStyles(Button)
                onClick={[Function]}
                variant="flat"
>
                signup
</WithStyles(Button)>,
              <WithStyles(Button)
                onClick={[Function]}
                variant="flat"
>
                Forget Password
</WithStyles(Button)>,
            ],
            "className": "flex mt3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Ok",
                "id": "ok",
                "onClick": [Function],
                "variant": "raised",
              },
              "ref": null,
              "rendered": "Ok",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "signup",
                "onClick": [Function],
                "variant": "flat",
              },
              "ref": null,
              "rendered": "signup",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Forget Password",
                "onClick": [Function],
                "variant": "flat",
              },
              "ref": null,
              "rendered": "Forget Password",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Paper)
          className="paperIn"
>
          <h4
                    className="mv3"
          >
                    Login
          </h4>
          <div
                    className="flex flex-column"
          >
                    <TextField
                              id="email"
                              label="Your email address"
                              onChange={[Function]}
                              required={false}
                              select={false}
                              type="text"
                              value=""
                    />
                    <TextField
                              id="password"
                              label="Password"
                              onChange={[Function]}
                              required={false}
                              select={false}
                              type="password"
                              value=""
                    />
          </div>
          <div
                    className="flex mt3"
          >
                    <WithStyles(Button)
                              id="ok"
                              onClick={[Function]}
                              variant="raised"
                    >
                              Ok
                    </WithStyles(Button)>
                    <WithStyles(Button)
                              onClick={[Function]}
                              variant="flat"
                    >
                              signup
                    </WithStyles(Button)>
                    <WithStyles(Button)
                              onClick={[Function]}
                              variant="flat"
                    >
                              Forget Password
                    </WithStyles(Button)>
          </div>
          <SnackBarCustom />
</WithStyles(Paper)>,
        "className": "paperOut",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h4
              className="mv3"
>
              Login
</h4>,
            <div
              className="flex flex-column"
>
              <TextField
                            id="email"
                            label="Your email address"
                            onChange={[Function]}
                            required={false}
                            select={false}
                            type="text"
                            value=""
              />
              <TextField
                            id="password"
                            label="Password"
                            onChange={[Function]}
                            required={false}
                            select={false}
                            type="password"
                            value=""
              />
</div>,
            <div
              className="flex mt3"
>
              <WithStyles(Button)
                            id="ok"
                            onClick={[Function]}
                            variant="raised"
              >
                            Ok
              </WithStyles(Button)>
              <WithStyles(Button)
                            onClick={[Function]}
                            variant="flat"
              >
                            signup
              </WithStyles(Button)>
              <WithStyles(Button)
                            onClick={[Function]}
                            variant="flat"
              >
                            Forget Password
              </WithStyles(Button)>
</div>,
            <SnackBarCustom />,
          ],
          "className": "paperIn",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "mv3",
            },
            "ref": null,
            "rendered": "Login",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextField
                  id="email"
                  label="Your email address"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="text"
                  value=""
/>,
                <TextField
                  id="password"
                  label="Password"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="password"
                  value=""
/>,
              ],
              "className": "flex flex-column",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": "email",
                  "label": "Your email address",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "id": "password",
                  "label": "Password",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(Button)
                  id="ok"
                  onClick={[Function]}
                  variant="raised"
>
                  Ok
</WithStyles(Button)>,
                <WithStyles(Button)
                  onClick={[Function]}
                  variant="flat"
>
                  signup
</WithStyles(Button)>,
                <WithStyles(Button)
                  onClick={[Function]}
                  variant="flat"
>
                  Forget Password
</WithStyles(Button)>,
              ],
              "className": "flex mt3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Ok",
                  "id": "ok",
                  "onClick": [Function],
                  "variant": "raised",
                },
                "ref": null,
                "rendered": "Ok",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "signup",
                  "onClick": [Function],
                  "variant": "flat",
                },
                "ref": null,
                "rendered": "signup",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Forget Password",
                  "onClick": [Function],
                  "variant": "flat",
                },
                "ref": null,
                "rendered": "Forget Password",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Signup 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withApollo(Apollo(Signup)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
